pandas/core/categorical.py

```diff
@@ -478,9 +534,10 @@ def _set_categories(self, categories, fastpath=False):
         """
 
         categories = self._validate_categories(categories, fastpath=fastpath)
-        if not fastpath and not self._categories is None and len(categories) != len(self._categories):
-            raise ValueError("new categories need to have the same number of items than the old "
-                             "categories!")
+        if (not fastpath and self._categories is not None and
+           len(categories) != len(self._categories)):
```

Comment:
It would be better to indent this 4 spaces beyond the indentation level of the `raise` on the next line.